### PR TITLE
WebAssembly port

### PR DESCRIPTION
**Pull request by kumpera**
This has the initial set of commits used by my hackweek port.
This includes configure.ac changes and the initial utils port.

### CODE REVIEW

File: mono/utils/mono-dl-wasm.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		".wasm",
+		"",
+	};
+	return suffixes;
+}
+
+int
+mono_dl_get_executable_path (char *buf, int buflen)
+{
+	strncpy (buf, "/managed", buflen);
```
Comment (alexrp) [inline mono/utils/mono-dl-wasm.c]: Should probably document the meaning of this string.

Comment (kumpera) [inline mono/utils/mono-dl-wasm.c]: Yeah, probably. Good thing it's not particularly useful, given dynamic loading is not supported at this stage of the port.

File: configure.ac [hunk 2]
```diff
@@ -96,8 +96,15 @@ need_link_unlink=no
 #Set to extra linker flags to be passed to the runtime binaries (mono /mono-sgen)
 extra_runtime_ldflags=""
 
+
+# Hack for WASM
+# Current autotools (v1.15) doesn't have a triplet we can use for wasm so the kludge we do is to
+# work around it by using a feature flag instead
+AC_ARG_ENABLE(wasm,[  --enable-wasm	Hack to set the current runtime to target wasm], enable_wasm=$enableval)
+
```
Comment (migueldeicaza) [inline configure.ac]: Since this is user-visible change, could we update README.md as well to list this new option?

Comment (kumpera) [inline configure.ac]: We'll probably have to change this later to be `--enable-host-wasm`  `--enable-target-wasm` to support AOT.

Comment (kumpera) [inline configure.ac]: Anyhow, documented the current one.

File: mono/utils/mono-dl-wasm.c (hunk 1 above)
Comment (kumpera) [inline mono/utils/mono-dl-wasm.c]: Fixed

**Pull request merged**